lib/chewy/strategy/lazy_sidekiq.rb

```diff
@@ -0,0 +1,58 @@
+module Chewy
+  class Strategy
+    # The strategy works the same way as sidekiq, but performs
+    # async evaluation of all index callbacks on model create and update
+    # driven by sidekiq
+    #
+    #   Chewy.strategy(:lazy_sidekiq) do
+    #     User.all.map(&:save) # Does nothing here
+    #     Post.all.map(&:save) # And here
+    #     # It schedules import of all the changed users and posts right here
+    #   end
+    #
+    class LazySidekiq < Sidekiq
+      class IndicesUpdateWorker
+        include ::Sidekiq::Worker
+
+        def perform(models)
+          Chewy.strategy(:sidekiq) do
```

Comment:
What it result in one more job scheduled? Wouldn't it be better to use `urgent` strategy here?

Reply:
@barthez, good point. Initially I made it intuitively as an offspring of `sidekiq` strategy. Actually I don't have a strong opinion on which strategy to use in `LazySidekiq::IndicesUpdateWorker`, the only thing that comes up to my mind is that `sidekiq` can be more performant due to multiple threads.